### ISSUE TITLE
Confusion in class OutlookAttention moduel

### ISSUE THREAD

**Issue by axhiao** [sail-sg/volo#12]
in `class OutlookAttention`, there is `self.v = nn.Linear(dim, dim, bias=qkv_bias)` and the input of this class is `x` whose shape is `B, H, W, C = x.shape`. My quesion is how this code `v = self.v(x).permute(0, 3, 1, 2)  # B, C, H, W` can go well without exception because matrix multiplication `[B, H, W, C] * [dim, dim]` will do here. And also in the original paper, Algorithm 1 implements `v_pj = nn.Linear(C, C)`. But in your codes, `C` is replaced with `dim`.  Thanks!

**Comment by yuanli2333** [comment]
dim in Transformer and Outlooker is the channel C, so dim=C.

**Comment by WilyZhao8** [comment]
> in `class OutlookAttention`, there is `self.v = nn.Linear(dim, dim, bias=qkv_bias)` and the input of this class is `x` whose shape is `B, H, W, C = x.shape`. My quesion is how this code `v = self.v(x).permute(0, 3, 1, 2) # B, C, H, W` can go well without exception because matrix multiplication `[B, H, W, C] * [dim, dim]` will do here. 

I'm also confused about this code. Looking forward to the author's explanation.
